tests/meta/clean/conftest.py

```diff
@@ -1,59 +1,139 @@
-def correct_flow_cell_path(
-    tmp_demultiplexed_runs_directory: Path, correct_flow_cell_name: str
-) -> Path:
-    """Full path to a correctly named flow cell directory in demultiplexed-runs."""
-    return Path(tmp_demultiplexed_runs_directory, correct_flow_cell_name)
+def housekeeper_api_with_flow_cell_to_clean(
+    real_housekeeper_api: HousekeeperAPI,
+    helpers: StoreHelpers,
+    hk_flow_cell_to_clean_bundle: Dict,
+    hk_sample_bundle_for_flow_cell_to_clean: Dict,
+):
+    """
+    Return a housekeeper api that contains a flow cell bundle with sample sheet,
+    a sample bundle with a fastq and a SPRING file that are tagged with the flow cell.
+    """
+    helpers.ensure_hk_bundle(store=real_housekeeper_api, bundle_data=hk_flow_cell_to_clean_bundle)
+    helpers.ensure_hk_bundle(
+        store=real_housekeeper_api, bundle_data=hk_sample_bundle_for_flow_cell_to_clean
+    )
+    return real_housekeeper_api
 
 
 @pytest.fixture(scope="function")
-def incorrect_flow_cell_path_too_long(
-    tmp_demultiplexed_runs_directory: Path, incorrect_flow_cell_too_long: str
-) -> Path:
-    """Full path to an incorrectly named flow cell directory in demultiplexed-runs."""
-    return Path(tmp_demultiplexed_runs_directory, incorrect_flow_cell_too_long)
+def hk_flow_cell_to_clean_bundle(
+    tmp_flow_cell_to_clean: FlowCellDirectoryData,
+    timestamp_yesterday: datetime,
+    tmp_sample_sheet_clean_flow_cell_path,
+) -> Dict:
+    """Housekeeper bundle information for a flow cell that can be cleaned."""
+    return {
+        "name": tmp_flow_cell_to_clean.id,
+        "created": timestamp_yesterday,
+        "expires": timestamp_yesterday,
+        "files": [
+            {
+                "path": str(tmp_sample_sheet_clean_flow_cell_path),
+                "archive": False,
+                "tags": ["samplesheet", tmp_flow_cell_to_clean.id],
+            }
+        ],
+    }
 
 
 @pytest.fixture(scope="function")
-def incorrect_flow_cell_path_extension(
-    tmp_demultiplexed_runs_directory: Path, incorrect_flow_cell_extension: str
-) -> Path:
-    """Full path to an incorrectly named flow cell directory in demultiplexed-runs."""
-    return Path(tmp_demultiplexed_runs_directory, incorrect_flow_cell_extension)
-
-
-@pytest.fixture(name="non-existent_flow_cell_path", scope="function")
-def non_existent_flow_cell_path(
-    tmp_demultiplexed_runs_directory: Path, incorrect_flow_cell_name: str
-) -> Path:
-    """Full path to an incorrectly named flow cell directory in demultiplexed-runs."""
-    return Path(tmp_demultiplexed_runs_directory, incorrect_flow_cell_name)
+def hk_sample_bundle_for_flow_cell_to_clean(
+    sample_id: str, timestamp_yesterday: datetime, spring_file, fastq_file, tmp_flow_cell_to_clean
+) -> Dict:
+    return {
+        "name": sample_id,
+        "created": timestamp_yesterday,
+        "expires": timestamp_yesterday,
+        "files": [
+            {
+                "path": spring_file.as_posix(),
```

Comment:
Should we also add the spring meta data file?

Reply:
I added the spring meta data as a requirement